components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/action/PreIssueAccessTokenResponseProcessor.java

```diff
@@ -113,7 +115,29 @@ public ActionExecutionStatus processSuccessResponse(Map<String, Object> eventCon
     private void logOperationExecutionResults(ActionType actionType,
                                               List<OperationExecutionResult> operationExecutionResultList) {
 
-        //todo: need to add to diagnostic logs
+        if (LoggerUtils.isDiagnosticLogsEnabled()) {
+
+            List<Map<String, String>> operationDetailsList = new ArrayList<>();
+            operationExecutionResultList.forEach(performedOperation -> {
+                operationDetailsList.add(Map.of(
+                        "operation", performedOperation.getOperation().getOp() + " with path: " +
+                                performedOperation.getOperation().getPath(),
+                        "status", performedOperation.getStatus().toString(),
+                        "message", performedOperation.getMessage()
+                ));
+            });
+
+            DiagnosticLog.DiagnosticLogBuilder diagLogBuilder = new DiagnosticLog.DiagnosticLogBuilder(
+                    "action-execution",
+                    "execute-action-operations");
+            diagLogBuilder
+                    .configParam("executed operations", operationDetailsList.isEmpty() ? "empty" : operationDetailsList)
```

Comment:
Let's make this input param. Because this is the response received and operations performed

Reply:
Updated in https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/2596/commits/6ce19cdf5e93cea123fcaff3796cb8418427937b